apps/hash-ai-worker-ts/src/activities/flow-activities/research-entities-action/infer-entities-from-web-page-worker-agent/tool-definitions.ts

```diff
@@ -0,0 +1,394 @@
+        description: {
+          type: "string",
+          description: dedent(`
+            A detailed description of the relevant information you are seeking from the PDF document.
+            Include keywords, phrases, or the name specific sections you are looking for.
+            Use language and terminology that is similar to the content in the PDF document you are seeking.
+            If you are looking for specific properties of an entity, name them.
+          `),
+        },
+        exampleText: {
+          type: "string",
+          description: dedent(`
+            An example of the data you are looking for in the PDF document.
+            This can be a table, a paragraph, or any other relevant content.
+          `),
+        },
+      },
+      required: ["fileUrl", "description", "explanation", "exampleText"],
+    },
+  },
+  /**
+   * @todo: consider unifying this with the `inferEntitiesFromWebPage` tool call,
+   * ensuring there are no regressions in the agent's ability to infer entities
+   * from paginated tables.
+   */
+  inferEntitiesFromText: {
+    name: "inferEntitiesFromText",
+    description: "Infer entities and links from text.",
+    inputSchema: {
+      type: "object",
+      properties: {
+        explanation: {
+          type: "string",
+          description: dedent(`
+            An explanation of why this tool call is required to satisfy the task,
+            and how it aligns with the current plan.
+
+            Provide your step by step thinking for why the provided text is sufficient
+              to infer the required entities and links to satisfy the research task.
+          `),
+          // Potentially also ask why other text is unsuitable
+        },
+        fileUrl: {
+          type: "string",
+          description: "The absolute URL of the file where the text is from.",
+        },
+        text: {
+          type: "string",
+          description: dedent(`
+            The text to infer entities and links from.
+
+            Include any relevant sections, paragraphs, tables, or other content that describe entities of the requested type(s).
+
+            When including a table, you must include table headers and any other relevant information to help the inference agent understand the data.
+
+            Do not under any circumstance truncate or provide partial text which may lead to missed entities
+              or properties.
+
+            You must provide as much text as necessary to infer all
+              the required entities and their properties from the web page in a single tool call.
+            `),
+        },
+        validAt: {
+          type: "string",
+          format: "date-time",
+          description: dedent(`
+            A date-time string in ISO 8601 format, representing when the provided text content is valid at.
+            If you don't know, assume it is the current date and time.
+            The current time is "${new Date().toISOString()}".
```

Comment:
I appreciate this is pre-existing but haven't looked at it again (see other comment) I'm not sure we should have an assumption that content is valid `now`... in many cases the date won't be inferrable from whatever bit of text the agent is looking at, and we will be giving false confidence that it is current

Reply:
Fair enough, I think the reason I added this a while ago was to satisfy the "measured at" property on the "Stock Constituent" entity type, where I thought if a date isn't specified on the web page, we can assume the current time is when the data should be assumed as correct. In the context of the FTSE constituents, that would be correct, but appreciate this might cause false confidence in other use-cases.

Should I remove the default here for now, and allow the agent to optionally pass it? In practice this will likely mean it never passes anything for this parameter, but I expect it will need to be reworked when we are passing facts instead of arbitrary text (where each fact may have a different `validAt` timestamp associated with it, depending on the context, or none if we don't know).

Reply:
We can leave it since you'll be reworking all this anyway, but I think we should generally not default to any assumptions about when data is current if it can't be inferred